src/framework/preferences/js/ArrowScrolling.js

```diff
@@ -0,0 +1,119 @@
+*/
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /************************************************************************************
+     * Scrolling Panel Prefs Editor:                                                    *
+     * This is a mixin grade to be applied to a fluid.prefs.prefsEditor type component. *
+     * Typically used for responsive small screen presentations of the separated panel  *
+     * to allow for scrolling by clicking on left/right arrows                          *
+     ************************************************************************************/
+
+    fluid.defaults("fluid.prefs.arrowScrolling", {
+        gradeNames: ["fluid.modelComponent"],
+        selectors: {
+            // panels: "", // should be supplied by the fluid.prefs.prefsEditor grade.
+            scrollContainer: ".flc-prefsEditor-scrollContainer"
+        },
+        model: {
+            // panelMaxIndex: null, // determined by the number of panels calculated after the onPrefsEditorMarkupReady event fired
+            // scrollToIndex: null, // the raw index set by translateToScroll, will be transformed to the panelIndex
+            panelIndex: 0
+        },
+        events: {
+            // beforeReset: null // should be provided by the fluid.prefs.prefsEditor grade
+        },
+        modelRelay: {
+            target: "panelIndex",
+            forward: {excludeSource: "init"},
+            singleTransform: {
+                type: "fluid.transforms.limitRange",
+                input: "{that}.model.scrollToIndex",
+                min: 0,
+                max: "{that}.model.panelMaxIndex"
+            }
+        },
+        modelListeners: {
+            "panelIndex": {
+                listener: "fluid.prefs.arrowScrolling.scrollToPanel",
+                args: ["{that}", "{change}.value"]
+            }
+        },
+        listeners: {
+            "onReady.windowResize": {
+                "this": window,
+                method: "addEventListener",
+                args: ["resize", "{that}.events.onSignificantDOMChange.fire"]
+            },
+            // Need to set panelMaxIndex after onPrefsEditorMarkupReady to ensure that the template has been
+            // rendered before we try to get the number of panels.
+            "onPrefsEditorMarkupReady.setPanelMaxIndex": {
+                changePath: "panelMaxIndex",
+                value: {
+                    expander: {
+                        funcName: "fluid.prefs.arrowScrolling.calculatePanelMaxIndex",
+                        args: ["{that}.dom.panels"]
+                    }
+                }
+            },
+            "beforeReset.resetPanelIndex": {
+                listener: "{that}.applier.fireChangeRequest",
+                args: {path: "panelIndex", value: 0, type: "ADD", source: "reset"}
+            }
+        },
+        invokers: {
+            scrollToPanel: {
```

Comment:
This invoker can be axed - the documentation that "scrollToIndex" is modelised is sufficient. Also it confusingly shares a name with the panelIndex listener